src/content.js

```diff
@@ -148,58 +202,116 @@ Function used to avoid multiple injection (cleaner than using an if?)
     */
     async function processImage(img) {
         debug('PROCESSING', img);
-        // This is the entire image size (should be atleas 10k pixels)
-        if ( img.width*img.height < 100*100 ) {
-            info('image too small', img.width, img.height);
+        
+        // Skip if already processed successfully
+        if (processedImages.has(img)) {
+            debug('image already processed successfully, skipping', img);
             return;
         }
         
-        // Get a blob from the image
-        const base64data = await base64FromAny(img);
+        // Skip if currently being processed to avoid concurrent processing
+        if (processingImages.has(img)) {
+            debug('image currently being processed, skipping', img);
+            return;
+        }
+        
+        // Mark as being processed
+        processingImages.add(img);
+        
+        try {
+            // Wait for image to load if not already loaded
+            if (img.tagName === 'IMG' && !img.complete) {
+                debug('waiting for image to load', img);
+                await new Promise((resolve) => {
+                    const onLoad = () => {
+                        img.removeEventListener('load', onLoad);
+                        img.removeEventListener('error', onLoad);
+                        resolve();
+                    };
+                    img.addEventListener('load', onLoad);
+                    img.addEventListener('error', onLoad);
+                    // Fallback timeout
+                    setTimeout(resolve, 5000);
```

Comment:
```suggestion
                    setTimeout(resolve, 15000);
```

Run this every 15 sec just in case